hm21-jpql

surdav opened the pull request.

surdav committed:
hm21-jpql

petrelevich commented on hm21-jpql/src/main/java/ru/otus/crm/model/Client.java:
```diff
@@ -0,0 +1,104 @@
+        this.phones = new ArrayList<>();
+        if (phones != null) {
+            for (Phone phone : phones) {
+                phone.setClient(this);
```
мутировать переданные аргументы не очень хорошо.
Лучше делать копию.

surdav committed:
Fix entity construction to avoid parameter mutation

- Modify Client constructor to create copies of Address and Phone objects instead of mutating input parameters
- Create deep copies of associated entities to maintain parameter immutability
- Prevent unexpected side effects when input objects are used elsewhere
- Follow the "Do not mutate input parameters" principle
